src/@next/pages/CheckoutPage/CheckoutPage.tsx

```diff
@@ -309,6 +370,59 @@ const CheckoutPage: React.FC<IProps> = ({}: IProps) => {
       break;
   }
 
+  useEffect(() => {
+    if (
+      !submitInProgress &&
+      checkout &&
+      location.pathname === "/checkout/payment-confirm" &&
+      !paymentConfirmation
+    ) {
+      handlePaymentConfirm();
+    }
+  }, [location.pathname, querystring, submitInProgress, checkout]);
+
+  const handlePaymentConfirm = async () => {
+    setSubmitInProgress(true);
+    setPaymentConfirmation(true);
+    if (querystring.resultCode === "Authorised") {
+      const { data, dataError } = await completeCheckout();
+      const errors = dataError?.error;
+      setSubmitInProgress(false);
+      if (errors) {
+        setPaymentGatewayErrors(errors);
+        const paymentStepLink = steps.find(
+          step => step.step === CheckoutStep.Payment
+        )?.link;
+        if (paymentStepLink) {
+          history.push(paymentStepLink);
+        }
+      } else {
+        setPaymentGatewayErrors([]);
+        handleStepSubmitSuccess(CheckoutStep.Review, {
+          id: data?.order?.id,
+          orderNumber: data?.order?.number,
+          token: data?.order?.token,
+        });
+      }
+    } else {
+      setPaymentGatewayErrors([
+        {
+          message: intl.formatMessage({
+            defaultMessage: "Payment confirmation went wrong.",
```

Comment:
Does it present same error message, no matter what the cause of the error was? Gateways tend to return the corresponding code of error (card expired, wrong data, fraud detected, etc).

Reply:
Hmm, yes, indeed we can use error codes returned by Adyen - https://docs.adyen.com/checkout/drop-in-web#step-6-present-payment-result - it is not very specific, but I'll use it. 

Nevertheless, there is a small problem, as we do not have a specific reason why this error occurred, because for now, the backend doesn't pass in query params such a reason, but it could - https://docs.adyen.com/checkout/drop-in-web#refused-response (we have to wait until backend will pass `refusalReason` - I have talked with the backend team and they should implement it sometime in the future)